types/watermark-dom/index.d.ts

```diff
@@ -0,0 +1,53 @@
+// Type definitions for watermark-dom 2.3
+// Project: https://github.com/saucxs/watermark-dom (Does not have to be to GitHub, but prefer linking to a source code repository rather than to a project website.)
+// Definitions by: shenhaoliang <https://github.com/shlroland>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+/*~ If this module is a UMD module that exposes a global variable 'myLib' when
+ *~ loaded outside a module loader environment, declare that global here.
+ *~ Otherwise, delete this declaration.
+ */
+export as namespace watermarkDom;
+export default watermarkDom;
+
+/*~ You can declare types that are available via importing the module */
+export interface Settings {
```

Comment:
Why do you make the interface members non-optional and instead make the whole interface later in `init`/`load` `Partial`? It's not very ergonomic for your users and it increases compile times. You should instead make all of the interface members optional:
```ts
export interface Settings {
    watermark_id?: string;
    watermark_prefix?: string;
    watermark_txt?: string;
    watermark_x?: number;
    ...
}
```